examples/customresources/demos/complexmodule/moduletest/module_test.go

```diff
@@ -37,31 +38,45 @@ import (
 // needs to be added to the web service before it normally would be avalilable after completing
 // a config cycle.
 func TestComplexModule(t *testing.T) {
-	logger := logging.NewTestLogger(t)
-
-	// Modify the example config to run directly, without compiling the module first.
-	cfgFilename, port, err := modifyCfg(t, utils.ResolveFile("examples/customresources/demos/complexmodule/module.json"), logger)
-	test.That(t, err, test.ShouldBeNil)
-
-	serverPath, err := testutils.BuildTempModule(t, "web/cmd/server/")
-	test.That(t, err, test.ShouldBeNil)
-
-	// start the viam server with a temporary home directory so that it doesn't collide with
-	// the user's real viam home directory
-	testTempHome := t.TempDir()
-	server := pexec.NewManagedProcess(pexec.ProcessConfig{
-		Name:        serverPath,
-		Args:        []string{"-config", cfgFilename},
-		CWD:         utils.ResolveFile("./"),
-		Environment: map[string]string{"HOME": testTempHome},
-		Log:         true,
-	}, logger.AsZap())
-
-	err = server.Start(context.Background())
-	test.That(t, err, test.ShouldBeNil)
-	defer func() {
-		test.That(t, server.Stop(), test.ShouldBeNil)
-	}()
+	logger, observer := logging.NewObservedTestLogger(t)
+
+	var port int
+	success := false
+	for portTryNum := 0; portTryNum < 10; portTryNum++ {
+		// Modify the example config to run directly, without compiling the module first.
+		cfgFilename, portLocal, err := modifyCfg(t, utils.ResolveFile("examples/customresources/demos/complexmodule/module.json"), logger)
+		port = portLocal
+		logger.Infow("Port discovery.", "TryNum", portTryNum, "Port", port)
```

Comment:
This was the first test I instrumented and accidentally left this in. I actually think its kind of nice, but not so nice that I want to go back and add it to the other tests. I hope this minor inconsistency isn't problematic.

Reply:
I think I would remove it